src/ucp/core/ucp_ep.c

```diff
@@ -580,6 +580,10 @@ ucs_status_ptr_t ucp_ep_close_nb(ucp_ep_h ep, unsigned mode)
         return UCS_STATUS_PTR(UCS_ERR_INVALID_PARAM);
     }
 
+    /* remove pending slow-path progress in case it wasn't removed yet */
```

Comment:
It should be done only from 1st `ucp_worker_iface_err_handle_progress` call and new entries shouldn't be enqueued if ep is failed (`UCP_EP_FLAG_FAILED`)

Reply:
And at completion of `ucp_ep_flush_internal` called below to cover "force" close case when 1st err handler is queued and ep can be destroyed earlier.